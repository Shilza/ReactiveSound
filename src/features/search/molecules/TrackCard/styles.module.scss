
.container {
  display: flex;
  flex-direction: column;
  border-left: 4px solid rgba(0,0,0,.1);
  padding: 14px 20px 0 16px;
  width: 700px;
}

.actions {
  display: flex;
  color: #5d5e5f;
  fill: #4d4e4f;
}

%text {
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  max-width: 85%;
  overflow: hidden;
}

.username {
  @extend %text;
  color: #7d7e7f;
  font-size: .8125rem;
}

.title {
  @extend %text;
  color: #bdbebf;
}