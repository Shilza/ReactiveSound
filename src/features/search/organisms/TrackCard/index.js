import React from "react";
import styles from './styles.module.scss';
import {Action} from "../../molecules";
import {Icon} from "../../../../ui/atoms";
import {Waveform} from "../../molecules";
import {connect} from "react-redux";
import {PlayButton} from "../../../common/atoms";
import {Button} from "../../../../ui/atoms";
import {Link} from "react-router-dom";
import {convertMsToString} from "../../../common/utils";

const TrackCard = ({id, onPlay, userId, link, isPlay, duration, title, dispatch, username, likesCount, playbackCount, waveform}) => (
    <div className={styles.container}>
        <Link to={`/users/${userId}/tracks`} className={styles.username}>{username}</Link>
        <div className={styles.title}>{title}</div>
        <Waveform id={id} waveform={waveform}/>
        <div className={styles.actions}>
            <Action label={convertMsToString(duration)}>
                <PlayButton id={id}/>
            </Action>
            <Action label={playbackCount}>
                <Icon name='headphones' viewBox="0 0 287.386 287.386"/>
            </Action>
            <Action label={likesCount}>
                <Icon name='like' viewBox="0 0 51.997 51.997"/>
            </Action>
            <Action>
                <a href={link} target='_blank' rel='noopener noreferrer'>
                    <Button>
                        <Icon name='chain' viewBox='0 0 500 500' fill='#4d4e4f'/>
                    </Button>
                </a>
            </Action>
        </div>
    </div>
);

export default connect(state => ({
    player: state.player.player,
    currentTrackId: state.player.currentTrack && state.player.currentTrack.id,
    trackIntervalId: state.player.trackIntervalId
}))(TrackCard);