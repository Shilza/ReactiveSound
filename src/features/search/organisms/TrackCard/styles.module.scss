@import '../../../../styles/utils.module';

.container {
  display: flex;
  flex-direction: column;
  border-left: 4px solid rgba(0,0,0,.1);
  padding: 14px 30px 0 30px;
  width: calc(100% - 260px);
}

.actions {
  display: flex;
  color: #5d5e5f;
  fill: color('accent');
}

%text {
  text-overflow: ellipsis;
  white-space: nowrap;
  word-wrap: normal;
  max-width: 85%;
  overflow: hidden;
}

.username {
  @extend %text;
  color: text('accent');
}

.title {
  @extend %text;
  color: text('title');
  margin-top: 5px;
}