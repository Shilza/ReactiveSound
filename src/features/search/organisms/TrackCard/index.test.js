import React from "react";
import {render} from 'react-testing-library';
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import TrackCard from './index';
import {BrowserRouter as Router} from "react-router-dom";

describe('Test search TrackCard', () => {
    const username = 'Username';
    const id = 5;
    const userId = 5;
    const playbackCount = 100;
    const link = 'link';
    const title = 'title';
    const duration = 200;
    const likesCount = 200;

    const mockStore = configureStore();
    const store = mockStore();

    const {container, getByTestId} = render(
        <Provider store={store}>
            <Router>
                <TrackCard
                    username={username}
                    id={id}
                    userId={userId}
                    playbackCount={playbackCount}
                    link={link}
                    title={title}
                    duration={duration}
                    likesCount={likesCount}
                />
            </Router>
        </Provider>
    );

    it('should renders correctly', () => {
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should have correct user\'s link', () => {
        const userLink = getByTestId('userLink');
        expect(userLink).toHaveAttribute('href', `/users/${userId}/tracks`);
    });

    it('should have correct track link', () => {
        const userLink = getByTestId('trackLink');
        expect(userLink).toHaveAttribute('href', link);
    });
});