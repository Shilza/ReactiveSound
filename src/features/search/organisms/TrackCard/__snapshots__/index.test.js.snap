// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test search TrackCard should renders correctly 1`] = `
<div
  class="container"
>
  <a
    class="username"
    data-testid="userLink"
    href="/users/5/tracks"
  >
    Username
  </a>
  <div
    class="title"
  >
    title
  </div>
  <div
    class="container"
  >
    <div
      class="loadingIndicator"
      data-testid="loader"
    >
      <div
        class="circle"
      />
      <div
        class="circle2"
      />
      <div
        class="circle3"
      />
    </div>
  </div>
  <div
    class="actions"
  >
    <div
      class="container"
    >
      <button
        aria-label="Pause"
        class="button"
        data-testid="playButton"
      >
        <svg
          fill="#4d4e4f"
          focusable="false"
          height="1em"
          role="img"
          viewBox="0 0 420 420"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.5,357h102V0h-102V357z M229.5,0v357h102V0H229.5z"
          />
        </svg>
      </button>
      <span
        class="label"
        data-testid="actionLabel"
      >
        00:00
      </span>
    </div>
    <div
      class="container"
    >
      <svg
        fill="#7f8081"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 287.386 287.386"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M62.743,155.437v98.42c0,5.867,4.741,10.605,10.605,10.605c5.854,0,10.605-4.738,10.605-10.605v-98.42
 c0-5.856-4.751-10.605-10.605-10.605C67.484,144.832,62.743,149.576,62.743,155.437z"
        />
        <path
          d="M29.456,264.582h23.351v-116.85c0.064-0.56,0.166-1.119,0.166-1.693c0-50.412,40.69-91.42,90.698-91.42
 c50.002,0,90.692,41.008,90.692,91.42c0,0.771,0.113,1.518,0.228,2.263v116.28h23.354c16.254,0,29.442-13.64,29.442-30.469
 v-60.936c0-13.878-8.989-25.57-21.261-29.249c-1.129-66.971-55.608-121.124-122.45-121.124
 c-66.86,0-121.347,54.158-122.465,121.15C8.956,147.638,0,159.32,0,173.187v60.926C0,250.932,13.187,264.582,29.456,264.582z"
        />
        <path
          d="M203.454,155.437v98.42c0,5.867,4.748,10.605,10.604,10.605s10.604-4.738,10.604-10.605v-98.42
 c0-5.856-4.748-10.605-10.604-10.605C208.191,144.832,203.454,149.576,203.454,155.437z"
        />
      </svg>
      <span
        class="label"
        data-testid="actionLabel"
      >
        100
      </span>
    </div>
    <div
      class="container"
    >
      <svg
        fill="#7f8081"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 51.997 51.997"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M51.911,16.242C51.152,7.888,45.239,1.827,37.839,1.827c-4.93,0-9.444,2.653-11.984,6.905
 c-2.517-4.307-6.846-6.906-11.697-6.906c-7.399,0-13.313,6.061-14.071,14.415c-0.06,0.369-0.306,2.311,0.442,5.478
 c1.078,4.568,3.568,8.723,7.199,12.013l18.115,16.439l18.426-16.438c3.631-3.291,6.121-7.445,7.199-12.014
 C52.216,18.553,51.97,16.611,51.911,16.242z M49.521,21.261c-0.984,4.172-3.265,7.973-6.59,10.985L25.855,47.481L9.072,32.25
 c-3.331-3.018-5.611-6.818-6.596-10.99c-0.708-2.997-0.417-4.69-0.416-4.701l0.015-0.101C2.725,9.139,7.806,3.826,14.158,3.826
 c4.687,0,8.813,2.88,10.771,7.515l0.921,2.183l0.921-2.183c1.927-4.564,6.271-7.514,11.069-7.514
 c6.351,0,11.433,5.313,12.096,12.727C49.938,16.57,50.229,18.264,49.521,21.261z"
        />
      </svg>
      <span
        class="label"
        data-testid="actionLabel"
      >
        200
      </span>
    </div>
    <div
      class="container"
    >
      <a
        data-testid="trackLink"
        href="link"
        rel="noopener noreferrer"
        target="_blank"
      >
        <button
          class="button"
        >
          <svg
            fill="#4d4e4f"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 500 500"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M312.453,199.601c-6.066-6.102-12.792-11.511-20.053-16.128c-19.232-12.315-41.59-18.859-64.427-18.859
 c-31.697-0.059-62.106,12.535-84.48,34.987L34.949,308.23c-22.336,22.379-34.89,52.7-34.91,84.318
 c-0.042,65.98,53.41,119.501,119.39,119.543c31.648,0.11,62.029-12.424,84.395-34.816l89.6-89.6
 c1.628-1.614,2.537-3.816,2.524-6.108c-0.027-4.713-3.87-8.511-8.583-8.484h-3.413c-18.72,0.066-37.273-3.529-54.613-10.581
 c-3.195-1.315-6.867-0.573-9.301,1.877l-64.427,64.512c-20.006,20.006-52.442,20.006-72.448,0
 c-20.006-20.006-20.006-52.442,0-72.448l108.971-108.885c19.99-19.965,52.373-19.965,72.363,0
 c13.472,12.679,34.486,12.679,47.957,0c5.796-5.801,9.31-13.495,9.899-21.675C322.976,216.108,319.371,206.535,312.453,199.601z"
            />
            <path
              d="M477.061,34.993c-46.657-46.657-122.303-46.657-168.96,0l-89.515,89.429c-2.458,2.47-3.167,6.185-1.792,9.387
 c1.359,3.211,4.535,5.272,8.021,5.205h3.157c18.698-0.034,37.221,3.589,54.528,10.667c3.195,1.315,6.867,0.573,9.301-1.877
 l64.256-64.171c20.006-20.006,52.442-20.006,72.448,0c20.006,20.006,20.006,52.442,0,72.448l-80.043,79.957l-0.683,0.768
 l-27.989,27.819c-19.99,19.965-52.373,19.965-72.363,0c-13.472-12.679-34.486-12.679-47.957,0
 c-5.833,5.845-9.35,13.606-9.899,21.845c-0.624,9.775,2.981,19.348,9.899,26.283c9.877,9.919,21.433,18.008,34.133,23.893
 c1.792,0.853,3.584,1.536,5.376,2.304c1.792,0.768,3.669,1.365,5.461,2.048c1.792,0.683,3.669,1.28,5.461,1.792l5.035,1.365
 c3.413,0.853,6.827,1.536,10.325,2.133c4.214,0.626,8.458,1.025,12.715,1.195h5.973h0.512l5.12-0.597
 c1.877-0.085,3.84-0.512,6.059-0.512h2.901l5.888-0.853l2.731-0.512l4.949-1.024h0.939c20.961-5.265,40.101-16.118,55.381-31.403
 l108.629-108.629C523.718,157.296,523.718,81.65,477.061,34.993z"
            />
          </svg>
        </button>
      </a>
      <span
        class="label"
        data-testid="actionLabel"
      />
    </div>
  </div>
</div>
`;
